driven/storage/adapter.go

```diff
@@ -1332,75 +1119,23 @@ func (sa *Adapter) CreateMemberUnchecked(clientID string, current *model.User, g
 
 // DeleteMember deletes a member membership from a specific group
 func (sa *Adapter) DeleteMember(clientID string, groupID string, userID string, force bool) error {
-	// transaction
-	err := sa.db.dbClient.UseSession(context.Background(), func(sessionContext mongo.SessionContext) error {
-		err := sessionContext.StartTransaction()
-		if err != nil {
-			log.Printf("error starting a transaction - %s", err)
-			return err
-		}
 
-		// get the member as we need to validate it
-		pipeline := []bson.M{
-			{"$unwind": "$members"},
-			{"$match": bson.M{"_id": groupID, "members.user_id": userID, "client_id": clientID}},
-		}
-		var result []struct {
-			ID     string `bson:"_id"`
-			Member member `bson:"members"`
-		}
-		err = sa.db.groups.AggregateWithContext(sessionContext, pipeline, &result, nil)
-		if err != nil {
-			abortTransaction(sessionContext)
-			return err
-		}
-		if result == nil || len(result) == 0 {
-			abortTransaction(sessionContext)
-			return errors.New("there is an issue processing the item")
-		}
-		resultItem := result[0]
-		member := resultItem.Member
-		if !(member.Status == "admin" || member.Status == "member") && !force {
-			abortTransaction(sessionContext)
-			return errors.New("you are not member/admin to the group")
-		}
-
-		//check if the member is admin, do not allow the group to become with 0 admins
-		if member.Status == "admin" && !force {
-			adminsCount, err := sa.findAdminsCount(sessionContext, groupID)
-			if err != nil {
-				abortTransaction(sessionContext)
-				return err
-			}
-			if *adminsCount == 1 {
-				abortTransaction(sessionContext)
-				return errors.New("you are the only admin for the group, you need to set another person for amdin before to leave")
+	currentMembership, _ := sa.FindGroupMembership(clientID, groupID, userID)
```

Comment:
This needs to be handled in a transaction to avoid race conditions

Reply:
Nice catch! Thanks!